store/store.go

```diff
@@ -1228,6 +1240,40 @@ var applyDelta = func(name string, deltaPath string, deltaInfo *snap.DeltaInfo)
 	return targetSnapPath, nil
 }
 
+// downloadAndApplyDelta downloads and then applies the delta to the current snap.
+func (s *Store) downloadAndApplyDelta(name, targetFn string, downloadInfo *snap.DownloadInfo, pbar progress.Meter, user *auth.UserState) (err error) {
+	deltaInfo := &downloadInfo.Deltas[0]
+	workingDir := filepath.Dir(targetFn)
+
+	deltaPath, err := s.downloadDelta(name, workingDir, downloadInfo, pbar, user)
```

Comment:
Happy to move to a different convention for `targetFn`, maybe `targetPath`? Open for suggestions. I picked the `fn` up from the review-tools and I like the brevity. But its indeed to easy to mix up with "function".